src/Health.Service/Reactive/ReactiveHealthCheck.cs

```diff
@@ -0,0 +1,57 @@
+﻿// -----------------------------------------------------------------------
+// <copyright file="ReactiveHealthCheck.cs" company="Payvision">
+//     Payvision Copyright © 2018
+// </copyright>
+// -----------------------------------------------------------------------
+
+using System.Collections.Generic;
+
+namespace Payvision.Diagnostics.Health.Reactive
+{
+    using System;
+    using System.Reactive.Concurrency;
+    using System.Reactive.Linq;
+
+    internal sealed class ReactiveHealthCheck : IHealthCheckFactory, IHealthCheckConfiguration
+    {
+        private readonly List<string> currentTags = new List<string>();
+
+        private IHealthCheck currentHealthCheck;
+
+        /// <inheritdoc />
+        public IHealthCheckConfiguration For(IHealthCheck healthCheck)
+        {
+            this.currentHealthCheck = healthCheck ?? throw new ArgumentNullException(nameof(healthCheck));
+            return this;
+        }
+
+        /// <inheritdoc />
+        public IHealthCheckConfiguration Tags(IEnumerable<string> tags)
+        {
+            this.currentTags.AddRange(tags);
+            return this;
+        }
+
+        internal IObservable<HealthCheckEntry> Build(IScheduler scheduler)
```

Comment:
I'm not sure but i think this check will only be executed ones per subscription not in a time base. 
Maybe we could clarify this point

Reply:
Actually we're building a promise here, which is going to be decorated on the health check set build method. Doing in that way we're going to be able of configure the execution behavior afterwards.